src/Compilers/CSharp/Portable/Lowering/AsyncRewriter/AsyncIteratorMethodToStateMachineRewriter.cs

```diff
@@ -271,13 +277,13 @@ public override BoundNode VisitYieldBreakStatement(BoundYieldBreakStatement node
 
             // Produce:
             //  disposeMode = true;
-            //  goto _enclosingFinallyOrExitLabel;
+            //  goto currentDisposalLabel;
 
             return F.Block(
```

Comment:
put a similar assertion: `Debug.Assert(CurrentDisposalLabel is object); // no yield return allowed inside a finally`